server/src/main/java/org/elasticsearch/cluster/InternalClusterInfoService.java

```diff
@@ -134,9 +134,15 @@ public void onMaster() {
         // Submit a job that will start after DEFAULT_STARTING_INTERVAL, and reschedule itself after running
         threadPool.scheduleUnlessShuttingDown(updateFrequency, executorName(), new SubmitReschedulingClusterInfoUpdatedJob());
 
-        if (clusterService.state().getNodes().getDataNodes().size() > 1) {
-            // Submit an info update job to be run immediately
-            threadPool.executor(executorName()).execute(() -> maybeRefresh());
+        try {
+            if (clusterService.state().getNodes().getDataNodes().size() > 1) {
+                // Submit an info update job to be run immediately
+                threadPool.executor(executorName()).execute(() -> maybeRefresh());
+            }
+        } catch (EsRejectedExecutionException ex) {
+            if (logger.isDebugEnabled()) {
```

Comment:
Given that this is an *extremely* rare case I think we can remove the guard here and make the code a bit simpler to read..

Reply:
Thanks makes sense. Addressed in 49ea80ff92b8fdeda8de39c5c292393691b5c576